### PR TITLE
YETUS-992. Create and publish a github action (more)

### PR DESCRIPTION
**Pull request by aw-was-here**
* use the github action to test the main tree in addition
  to the normal testing
* speed up the github action bits in use by caching maven
  and fix some related issues with using a pre-built cache
* add documentation about --ignore-unknown-options
* add --report-unknown-options support and docs

### CODE REVIEW

File: start-build-env.sh [hunk 1]
```diff
@@ -92,14 +92,26 @@ cd "${ROOTDIR}"
...
 
-if [[ -z ${GITHUB_ACTIONS} ]]; then
-  dockerargs+=(-v "${HOME}/.m2:/home/${USER_NAME}/.m2${V_OPTS:-}")
-  dockerargs+=(-v "${HOME}/.gnupg:/home/${USER_NAME}/.gnupg")
-  dockerargs+=(-v "${HOME}/.ssh:/home/${USER_NAME}/.ssh")
+# git opertions
+if [[ ! -d ${HOME}/.ssh ]]; then
+  mkdir "${HOME}/.ssh"
 fi
+dockerargs+=(-v "${HOME}/.ssh:/home/${USER_NAME}/.ssh${V_OPTS:-}")
```
Comment (ndimiduk) [inline start-build-env.sh:114]: There's not a better way to pass key materials into the docker process? I'd hope for some solution other than making the private keys visible to the process, something like agent forwarding.

**Review by ndimiduk — APPROVED** [review]
Skimmed, looks alright to me. Reviewing for changes in the face of sort-ordering lists was a bit challenging, a comment for next time.

**Comment by aw-was-here** [comment]
I screwed up the linkchecker tuning.  Fixed.  Shaved off 6 minutes. Hooray!

File: start-build-env.sh (hunk 1 above)
Comment (aw-was-here) [reply start-build-env.sh:124]: There are a lot of games one can play with the gpg-agent socket. We should probably look into them now that ssh login into github, etc, are more prevalent.  It should be noted that this code is in start-build-env.sh, which up until now was only used for local work.  In the GitHub actions case, there are no ssh creds to utilize.

**Comment by aw-was-here** [comment]
Thanks for the approval @ndimiduk !  merging

**Pull request merged**